help_test.go

```diff
@@ -512,3 +512,55 @@ Group 2
 		require.Equal(t, expected, w.String())
 	})
 }
+
+func TestUsageOnError(t *testing.T) {
+	var cli struct {
+		Flag string `help:"A required flag." required`
+	}
+	w := &strings.Builder{}
+	p := mustNew(t, &cli,
+		kong.Writers(w, w),
+		kong.Description("Some description."),
+		kong.Exit(func(int) {}),
+		kong.UsageOnError(),
+	)
+	_, err := p.Parse([]string{"--help"})
+	require.Error(t, err)
+	//p.FatalIfErrorf(err)
+
+	expected := `Usage: test --flag=STRING
+
+Some description.
+
+Flags:
+  -h, --help           Show context-sensitive help.
+      --flag=STRING    A required flag.
+`
+	require.Equal(t, expected, w.String())
+}
+
+func TestShortUsageOnError(t *testing.T) {
+	var cli struct {
+		Flag string `help:"A required flag." required`
+	}
+	w := &strings.Builder{}
+	p := mustNew(t, &cli,
+		kong.Writers(w, w),
+		kong.Description("Some description."),
+		kong.Exit(func(int) {}),
+		kong.ShortUsageOnError(),
+	)
+	_, err := p.Parse([]string{})
+	require.Error(t, err)
+	p.FatalIfErrorf(err)
+
+	expected := `Usage: test --flag=STRING
```

Comment:
Is this the kind of help you're aiming for? It's not as succinct as I was imagining! Feel free to adjust the output further if you prefer.

Reply:
I am not sure. What were you thinking? Without description it would be:
```
Usage: test --flag=STRING

Run "test --help" for more information.

test: error: missing flags: --flag=STRING
```
I think I would prefer:

```
Usage: test --flag=STRING
Run "test --help" for more information.

test: error: missing flags: --flag=STRING
```
but I'm not very opinionated here.

The reason I left the description in apart from it being the normal flow anyway is that I use the description together with `NoAppSummary: true` to get a custom Usage line highlighting the flags I want. I suppose that's a bit of a hack in the first place.

Do you think I should just add another HelpOption field called say `Short` and create a custom output for it?

Reply:
I honestly can't remember what we were trying to achieve! :) I think it was motivated by something you did in gurl? Let's have a chat about it on Monday?